python/lsst/obs/base/makeRawVisitInfo.py

```diff
@@ -105,6 +105,8 @@ def setArgDict(self, md, argDict):
         Subclasses should expand this or replace it.
         """
         argDict["exposureTime"] = self.popFloat(md, "EXPTIME")
+        darkTime = self.popFloat(md, "DARKTIME") if md.exists("DARKTIME") else NaN
+        argDict["darkTime"] = darkTime if not math.isnan(darkTime) else argDict["exposureTime"]
```

Comment:
I think it is not reasonable to have the default implementation first look for one keyword and then fall back to another. That makes it unsafe for subclasses to supplement this method by overriding it to do more work, then calling this to do the basics. As the doc string says, *all* subclasses are intended to override this method. However, I also intended that this method be useful for the basics in many cases, so the override could call into this. The policy of "use DARKTIME if found, else use EXPTIME" is not safe in cases where DARKTIME ought to be present but may be missing due to upstream problems.

Beyond that the implementation has several defects:
- If "DARKTIME" is present with a value of `NaN` (or a missing value, should we ever support that) then `darkTime` will be set to the value of "EXPTIME" instead of the correct value of `NaN`.
- The `if ...` in the first line does nothing;`popFloat` already returns `NaN` if the keyword is not found.

I am happy to have `darkTime = self.popFloat(md, "DARKTIME")` in the base class, but I am not happy to go any further than that. If you need more than please add it to subclasses, not this class.

Also, if you think this "if one keyword is missing then use another" is going to be a common idiom for real cameras please consider adding a method to handle it, since the logic is a bit tricky. If you'd rather make that a separate ticket I'll be happy to have a go at it. The nuisance is that we already have type-specific getters, and this adds a different dimension.